src/sentry/llm/providers/openai.py

```diff
@@ -0,0 +1,45 @@
+from typing import Any
+
+from openai import OpenAI
+
+from sentry.llm.providers.base import LlmModelBase
+
+
+class OpenAIProvider(LlmModelBase):
+
+    provider_name = "openai"
+
+    def _complete_prompt(
+        self,
+        *,
+        usecase_config: dict[str, Any],
+        prompt: str,
+        message: str,
+        temperature: float,
+        max_output_tokens: int,
+    ) -> str | None:
+        model = usecase_config["options"]["model"]
+        client = get_openai_client(self.provider_config["options"]["api_key"])
+
+        response = client.chat.completions.create(
+            model=model,
+            temperature=temperature
+            * 2,  # open AI temp range is [0.0 - 2.0], so we have to multiply by two
+            messages=[
+                {"role": "system", "content": prompt},
```

Comment:
if we are providing the prompt (ex. something like "return whether this is spam or not spam") should this technically be also `"role": "user"`? (don't think the role actually matters, this is more of a nit/small implementation detail)

Reply:
i think the idea of the system message is used correct here:
https://community.openai.com/t/what-exactly-does-a-system-msg-do/459409

> the user messages are messages that the user wrote
the assistant messages those the bot wrote
the system message is a message that the developer wrote to tell the bot how to interpret the conversation. They’re supposed to give instructions that can override the rest of the convo, but they’re not always super reliable depending on what model you’re using.

Reply:
ah makes sense, thanks for clarifying